### DISCUSSION TITLE
[RFC] New Variant: Anchor direction horizontal

### DISCUSSION THREAD

**Discussion by MadCcc** [ant-design/ant-design#38895] [RFCs - archive]
## Summary
Anchor 组件新增 `direction` 属性，支持 `vertical`，新增 `horizontal` 样式。

## Motivation
页面中经常会需要横向的 Anchor 导航，虽然有时会和 Tabs 混淆，但是功能上有很大差别（内容是否同时展示）。

## API

| 属性        | 说明     | 类型             | 默认值 |
| --- | --- | --- | --- |
| direction | 展示方向 | `vertical` \| `horizontal` | `vertical` | 

仅支持一级目录

## Basic Example
```tsx
import React from 'react';
import { Anchor } from 'antd';

const { Link } = Anchor;

const App: React.FC = () => (
  <Anchor direction="horizontal">
    <Link href="#components-anchor-demo-basic" title="Basic demo" />
    <Link href="#components-anchor-demo-static" title="Static demo" />
  </Anchor>
);

export default App;
```

### 样式
基本样式与 Tabs 一致。
![image](https://user-images.githubusercontent.com/27722486/203467365-16473e9f-7e5c-4332-a272-85cf13f16c86.png)

### 背景色
由于横向的 Anchor 一般用于纵向排版，所以背景色需要是白色，防止元素重叠。

## Drawbacks
按照设计和使用方法，仅支持一级目录。

## Bonus
Anchor 改为数据驱动，废弃 JSX 语法糖

**Comment by foryuki** [comment]
我想做这个~

**Comment by afc163** [comment]
这个在演示上要展现出区域滚动和固钉效果，不能直接放一个 Anchor，不然真的和 tabs 一模一样了。

**Comment by foryuki** [comment]
@MadCcc @zombieJ 
- 横向 anchor 容器是否需要和 Tabs 一样支持左右滑动容纳更多锚点？
- 横向 anchor 容器是否支持锚点过多时...形式缩略？
![image](https://user-images.githubusercontent.com/8649233/204124277-d8f631b4-d65a-489d-a3ec-e8f847eec684.png)

**Reply by MadCcc** [reply]
先只做简单 overflow，不显示 ... 菜单，支持左右滑动。

**Reply by foryuki** [reply]
好的👌

**Comment by foryuki** [comment]
## 计划
- 先做数据驱动改造，废弃 JSX 语法糖。 https://github.com/ant-design/ant-design/pull/39034
- 再实现横向 Anchor，横向 Anchor 只支持数据驱动。 https://github.com/ant-design/ant-design/pull/39372
